Crossdev commit 1ebd639ccc8605f3e69b5973788101264902559e added an extra
check to prevent overwriting secondary host toolchains. But we rely on
being able to do that in Chromium OS for i686-pc-linux-gnu.

This patch reverts the additional check so that we can continue using
crossdev to build i686-pc-linux-gnu. b/187786439 will remove using of
i686-pc-linux-gnu so that we can drop this patch.

diff --git a/crossdev b/crossdev
index 25b7acf..e1f7d01 100755
--- a/crossdev
+++ b/crossdev
@@ -495,7 +495,6 @@ show_repo_cfg() {
 setup_portage_vars() {
 	local arch=${ARCH} arch_set=${ARCH+set}
 	local chost=${CHOST} chost_set=${CHOST+set}
-	local abi
 	unset ARCH CHOST
 	eval $(portageq envvar -v PORT_LOGDIR PORTAGE_TMPDIR ARCH CHOST)
 	parse_repo_config
@@ -506,17 +505,6 @@ setup_portage_vars() {
 	HCHOST=${CHOST}
 	[[ ${chost_set} == "set" ]] && CHOST=${chost} || unset CHOST
 
-	# Collect all CHOST* visible for host. We need to make sure
-	# not to clobber any of native CHOSTs.
-	HCHOSTS=(${HCHOST})
-	for abi in $(portageq envvar ABI MULTILIB_ABIS DEFAULT_ABI); do
-		HCHOSTS+=($(portageq envvar CHOST_${abi}))
-	done
-	# uniq
-	HCHOSTS=(
-		$(printf "%s\n" "${HCHOSTS[@]}" | sort -u)
-	)
-
 	# see if user told us where to write things, otherwise
 	# set up some repos.conf magic if possible.  if not,
 	# install our stuff to the first overlay in the list.
@@ -771,7 +759,6 @@ UOPTS=
 TARCH=
 HARCH=
 HCHOST=
-HCHOSTS=()
 CTARGET=
 MULTILIB_ABIS="default"
 MULTILIB_USE=""
@@ -962,15 +949,6 @@ if [[ ${HCHOST} == "${CTARGET}" ]] ; then
 	exit 1
 fi
 
-for hchost in "${HCHOSTS[@]}"; do
-	if [[ ${hchost} == "${CTARGET}" ]] ; then
-		eerror "Refusing to create a cross-compiler using the same"
-		eerror "target name as your host utils."
-		eerror "Consider using sys-devel/multilib-gcc-wrapper package."
-		exit 1
-	fi
-done
-
 # grab user settings
 d="${CONFIGROOT}/crossdev/${CTARGET}"
 for v in MULTILIB_ABIS USE BVER GVER KVER LVER STAGE CFLAGS LDFLAGS ASFLAGS ; do
@@ -1064,7 +1042,7 @@ info() {
 hr -
 einfo "crossdev version:      ${CROSSDEV_VER}"
 einfo "Host Portage ARCH:     ${HARCH}"
-einfo "Host Portage System:   ${HCHOST} (${HCHOSTS[*]})"
+einfo "Host Portage System:   ${HCHOST}"
 einfo "Target Portage ARCH:   ${TARCH}"
 einfo "Target System:         ${CTARGET}"
 einfo "Stage:                 ${STAGE} (${STAGE_DISP[${STAGE}]})"
-- 
2.33.1.1089.g2158813163f-goog

